test/test_nn.py

```diff
@@ -14564,6 +14564,48 @@ def test_maxpool3d_non_square_backward(self, device):
             F.max_pool3d(x, kernel_size=(1, 1, 1)).sum().backward()
             self.assertTrue(torch.allclose(x.grad, torch.ones_like(x.grad)))
 
+    # Check that clip_grad_norm_ raises an error if the norm of a gradient
+    # is non-finite
+    def test_clip_grad_norm_error_if_nonfinite(self, device):
+        test_cases = [
+            # scalar, is_grad_nonfinite
+            (inf, True),
+            (-inf, True),
+            (nan, True),
+            (2e22, False),
+            (-2e22, False),
```

Comment:
I think there should be a test case for finite values large enough to result in a non-finite total norm for the gradients.

The issue reporter provided an example of this (see https://github.com/pytorch/pytorch/issues/46849#issue-729701939). They use a tensor of dtype float32 and size 100 with each element equal to 1e20.

Reply:
Added